notebooks/Generic/demo_units_custom.ipynb

```diff
@@ -0,0 +1,334 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up two datasets with x-axes that use different spectral coordinates (which get auto-converted because of the WCSLink) and y-axes that use different temperature units (just to show how to add custom equivalencies)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs1 = WCS(naxis=1)\n",
+    "wcs1.wcs.ctype = ['WAVE']\n",
+    "wcs1.wcs.crval = [10]\n",
+    "wcs1.wcs.cdelt = [3]\n",
+    "wcs1.wcs.crpix = [1]\n",
+    "wcs1.wcs.cunit = ['cm']\n",
+    "\n",
+    "d1 = Data(temp1=np.random.random(10))\n",
+    "d1.get_component('temp1').units = 'deg_C'\n",
+    "d1.coords = wcs1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs1 = WCS(naxis=1)\n",
+    "wcs1.wcs.ctype = ['FREQ']\n",
+    "wcs1.wcs.crval = [1000]\n",
+    "wcs1.wcs.cdelt = [1000]\n",
+    "wcs1.wcs.crpix = [1]\n",
+    "wcs1.wcs.cunit = ['MHz']\n",
+    "\n",
+    "d2 = Data(temp2=np.random.random(10) + 273)\n",
+    "d2.get_component('temp2').units = 'K'\n",
+    "d2.coords = wcs1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up a custom unit conversion class with equivalencies:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy import units as u\n",
+    "from glue.core.units import unit_converter\n",
+    "\n",
+    "@unit_converter('custom-astropy')\n",
+    "class UnitConverterWithTemperature:\n",
+    "\n",
+    "    def equivalent_units(self, data, cid, units):\n",
+    "        equivalencies = u.temperature() if 'temp' in cid.label.lower() else None\n",
+    "        return map(u.Unit(units).find_equivalent_units(include_prefix_units=True, equivalencies=equivalencies), str)\n",
+    "\n",
+    "    def to_unit(self, data, cid, values, original_units, target_units):\n",
+    "        equivalencies = u.temperature() if 'temp' in cid.label.lower() else None\n",
+    "        return (values * u.Unit(original_units)).to_value(target_units, equivalencies=equivalencies)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use the converter class defined above"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from glue.config import settings\n",
+    "settings.UNIT_CONVERTER = 'custom-astropy'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/tom/Code/glue/glue/external/echo/__init__.py:3: UserWarning: glue.external.echo is deprecated, import from echo directly instead\n",
```

Comment:
You probably want to clear the outputs if you plan to keep this notebook at merge time.